File: scripts/wallet-setup.js
#!/usr/bin/env node
/**
 * wallet-setup.js — Create or load a JoyClaw EVM wallet
 * Saves to ~/.joyclaw/wallet.json (mode 0600)
 */
const fs = require('fs')
const path = require('path')

const dir = path.join(process.env.HOME, '.joyclaw')
const walletFile = path.join(dir, 'wallet.json')

fs.mkdirSync(dir, { recursive: true })

if (!fs.existsSync(walletFile)) {
  let ethers
  try {
    ethers = require('ethers')
  } catch {
    console.error('ERR: ethers not found. Run: npm install (in skill dir)')
    process.exit(1)
  }
  const wallet = ethers.Wallet ? ethers.Wallet.createRandom() : new ethers.Wallet(ethers.utils?.randomBytes?.(32) ?? require('crypto').randomBytes(32))
  const data = { address: wallet.address, privateKey: wallet.privateKey }
  fs.writeFileSync(walletFile, JSON.stringify(data, null, 2), { mode: 0o600 })
  console.log('CREATED')
  console.log('ADDRESS=' + wallet.address)
} else {
  const w = JSON.parse(fs.readFileSync(walletFile, 'utf8'))
  console.log('EXISTS')
  console.log('ADDRESS=' + w.address)
}

File: scripts/wallet-setup.py
#!/usr/bin/env python3
"""
wallet-setup.py — EVM wallet init for JoyClaw
Writes ~/.joyclaw/wallet.json if it doesn't exist.

Output (stdout):
  CREATED\nADDRESS=0x...   (first run)
  EXISTS\nADDRESS=0x...    (subsequent runs)
"""
import json
import os
import sys
from pathlib import Path

try:
    from eth_account import Account
except ImportError:
    os.system("pip install eth-account -q")
    from eth_account import Account

jc_dir = Path.home() / ".joyclaw"
jc_dir.mkdir(exist_ok=True)
wallet_file = jc_dir / "wallet.json"

if wallet_file.exists():
    data = json.loads(wallet_file.read_text())
    print("EXISTS")
    print(f"ADDRESS={data['address']}")
else:
    acct = Account.create()
    data = {"address": acct.address, "privateKey": acct.key.hex()}
    wallet_file.write_text(json.dumps(data, indent=2))
    wallet_file.chmod(0o600)
    print("CREATED")
    print(f"ADDRESS={acct.address}")
